util/collate/charset.go

```diff
@@ -20,6 +20,49 @@ import (
 	"github.com/pingcap/tidb/parser/charset"
 )
 
+// CharacterMode is TiDBRestrictCharacterMode value.
+type CharacterMode int
+
+// SkipUTF8Check return true if CharMode contains SKIP_UTF8_CHECK.
+func (m CharacterMode) SkipUTF8Check() bool {
+	return m&skipUTF8Check != 0
+}
+
+// EnableInvalidCharacter return true if CharMode contains ENABLE_INVALID_CHARACTER.
+func (m CharacterMode) EnableInvalidCharacter() bool {
+	return m&enableInvalidCharacter != 0
+}
+
+const (
+	enableInvalidCharacter CharacterMode = 1 << iota
+	skipUTF8Check
+)
+
+const (
+	// DefaultCharMode is the default value of TiDBRestrictCharacterMode.
+	DefaultCharMode = ""
+	// EnableInvalidCharacter enable the invalid character from the client.
+	EnableInvalidCharacter = "ENABLE_INVALID_CHARACTER"
```

Comment:
IMHO the names of 'ENABLE_INVALID_CHARACTER' and 'SKIP_UTF8_CHECK' are confusing.

'ENABLE_INVALID_CHARACTER' determines the behavior of 'converting' non-utf8 encodings, while 'SKIP_UTF8_CHECK' controls the behavior of invalid utf8 encodings when inserting the data.

I still prefer to have the names like 'ALLOW_INVALID_NON_UTF8' and 'ALLOW_INVALID_UTF8'.